### PR TITLE
feat(mantine, table): rework actions again

### PR DESCRIPTION
**Pull request by GermainBergeron**
### Proposed Changes

Rework the actions yet again, this time I think it covers all the cases

Potential discussion points:
- I had to add `getRowActions` and `getMultiSelectionRowActions` in the table context to make it accessible in the column
- I couldn't keep the same standard `(datum: TData, index: number, row: Row<TData>)` to render the actions, the Table.Header doesn't have access to the Row

### Potential Breaking Changes

Table.Actions is gone

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/mantine/src/components/table/table-actions/TableActionContext.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import {createSafeContext} from '@mantine/core';
+
+export interface TableActionContextValue {
+    primary: boolean;
+}
+
+export const [TableActionProvider, useTableActionContext] = createSafeContext<TableActionContextValue>(
```
Comment (gdostie) [inline packages/mantine/src/components/table/table-actions/TableActionContext.ts:7]: Where is this used?

**Review by toofff — COMMENTED** [review]
What were the problems of configuring actions with the compound component pattern?

File: packages/mantine/src/components/table/table-actions/TableActionContext.ts (hunk 1 above)
Comment (GermainBergeron) [reply packages/mantine/src/components/table/table-actions/TableActionContext.ts:7]: In the action to determine if it's primary or secondary: https://github.com/coveo/plasma/pull/3725/files#diff-fc82ca754653dc5b838fec56691ed3fe2538f00d1a544cb7d0b8d5b28fc61c96R43

And in the ActionList to wrap each section with the correct provider:
- https://github.com/coveo/plasma/pull/3725/files#diff-45d043921d99311e7caff64898eade5624e8c3ee09f64e8166eb3713146993d6R109-R111
- https://github.com/coveo/plasma/pull/3725/files#diff-45d043921d99311e7caff64898eade5624e8c3ee09f64e8166eb3713146993d6R140

It really simplified the code

**Comment by gdostie** [comment]
> What were the problems of configuring actions with the compound component pattern?

All the actions had to be rendered as a direct child of `<Table.Actions>` (or within a Fragment), so you could not create reusable actions or use the Guard component around the individual actions. Overall it was too limiting on the table implementer side.

File: packages/mantine/src/components/table/Table.tsx [hunk 2]
```diff
@@ -293,7 +289,6 @@ export const TableComponentsOrder = {
 };
 
 Table.AccordionColumn = TableAccordionColumn;
-Table.Actions = TableActions;
 Table.ActionsColumn = TableActionsColumn;
 Table.ActionItem = TableActionItem;
 Table.ActionLabel = Menu.Label;
```
Comment (toofff) [inline packages/mantine/src/components/table/Table.tsx:294]: Do all components always have to be accessible?
```
Table.ActionsColumn = TableActionsColumn;
Table.ActionItem = TableActionItem;
Table.ActionLabel = Menu.Label;
Table.ActionDivider = Menu.Divider;
```

Comment (toofff) [reply packages/mantine/src/components/table/Table.tsx:294]: Lines 308 and 309 are code duplication ;)

image

Comment (GermainBergeron) [reply packages/mantine/src/components/table/Table.tsx:294]: What do you mean? Are you talking about component like TableActionList? Those are internal components and shouldn't be used in the admin

**Review by toofff — COMMENTED** [review]
I find setting up the menu a little more complicated than before, but there are certainly good reasons for this.

Perhaps it would be interesting to add some simple examples to cover all cases of the new action menu setup?

File: packages/mantine/src/components/table/Table.tsx (hunk 2 above)
Comment (toofff) [reply packages/mantine/src/components/table/Table.tsx:294]: I see you've removed the `Table.Actions` component, but the others should still be present? :arrow_down: 

```
Table.ActionsColumn = TableActionsColumn;
Table.ActionItem = TableActionItem;
Table.ActionLabel = Menu.Label;
Table.ActionDivider = Menu.Divider;
```

Comment (GermainBergeron) [reply packages/mantine/src/components/table/Table.tsx]: I kept `Table.ActionsColumn` and `Table.ActionItem`, I removed the others

**Review by gdostie — APPROVED** [review]

**Review by toofff — APPROVED** [review]

**Pull request merged**